File: 2022.01.24/javascript/15652.js
const [n,m] = require('fs').readFileSync('/dev/stdin').toString().split('\n')[0].split(' ').map(Number);

const solution = (n,cnt,target,acc) => {
  if (cnt === target){
    return acc.join(" ") + "\n";
  }
  let result = ""
  for (let i = acc[acc.length-1] || cnt+1 ; i<=n ; i++){
    result += solution(n,cnt+1,target,acc.concat(i));
  }
  return result;
}
console.log(solution(n,0,m,[]));

File: 2022.01.24/python/15652.py
n,m = map(int,input().split())
def search(start,arr,n,m):
    cnt = len(arr)
    if cnt==m:
        print(" ".join(map(str,arr)))
        return
    for i in range(start,n+1):
        if i>=arr[-1]:
            search(i,arr+[i],n,m)
for i in range(1,n+1):
    search(i,[i],n,m)